pkg/alertmanager/multitenant_test.go

```diff
@@ -41,26 +46,64 @@ receivers:
 
 // basic easily configurable mock
 type mockAlertStore struct {
-	configs map[string]alerts.AlertConfigDesc
+	configs     map[string]alerts.AlertConfigDesc
+	OnList      func()
+	WithListErr error
 }
 
-func (m *mockAlertStore) ListAlertConfigs(ctx context.Context) (map[string]alerts.AlertConfigDesc, error) {
+func (m *mockAlertStore) ListAlertConfigs(_ context.Context) (map[string]alerts.AlertConfigDesc, error) {
+	if m.OnList != nil {
+		m.OnList()
+	}
+
+	if m.WithListErr != nil {
+		return nil, m.WithListErr
+	}
+
 	return m.configs, nil
 }
 
-func (m *mockAlertStore) GetAlertConfig(ctx context.Context, user string) (alerts.AlertConfigDesc, error) {
+func (m *mockAlertStore) GetAlertConfig(_ context.Context, _ string) (alerts.AlertConfigDesc, error) {
 	return alerts.AlertConfigDesc{}, fmt.Errorf("not implemented")
 }
 
-func (m *mockAlertStore) SetAlertConfig(ctx context.Context, cfg alerts.AlertConfigDesc) error {
+func (m *mockAlertStore) SetAlertConfig(_ context.Context, cfg alerts.AlertConfigDesc) error {
 	m.configs[cfg.User] = cfg
 	return nil
 }
 
-func (m *mockAlertStore) DeleteAlertConfig(ctx context.Context, user string) error {
+func (m *mockAlertStore) DeleteAlertConfig(_ context.Context, _ string) error {
 	return fmt.Errorf("not implemented")
 }
 
+func mockAlertmanagerConfig(t *testing.T) *MultitenantAlertmanagerConfig {
+	t.Helper()
+
+	externalURL := flagext.URLValue{}
+	err := externalURL.Set("http://localhost/api/prom")
+	require.NoError(t, err)
+
+	tempDir, err := ioutil.TempDir(os.TempDir(), "alertmanager")
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		err := os.RemoveAll(tempDir)
+		require.NoError(t, err)
+	})
+
+	cfg := &MultitenantAlertmanagerConfig{
+		PollInterval: time.Minute,
```

Comment:
I think you need to set this after calling `flagext.DefaultValues()`. Doesn't `DefaultValues()` override it?